bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -504,8 +504,8 @@ public void added(Provider<Rule> provider, Rule element) {
             String rUID = ruleEngine.getUniqueId();
             ruleWithUID = initRuleId(rUID, element);
         }
-        super.added(provider, ruleWithUID);
```

Comment:
Is it safe to change this order? 
You are now sending the event, before the rule is available in the registry - why is this necessary?